### ISSUE TITLE
pulse_lib.continuous pulses do not handle list inputs

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-terra#2475]
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
```python

from qiskit.pulse.pulse_lib.continuous import sin
sin([0,0.1,0.1,0.3], amp=1, freq=0.01)
```
gives

```python
TypeError: can't multiply sequence by non-int of type 'float'
```

### Suggested solutions
The use of `ndarray` in typing is not really supported; See for example: https://github.com/numpy/numpy/issues/7370.  So a list just passes through.  There should be an `np.asarray()` call on the input `times` arg.  This is true for all the pulses in the continuous module.

**Comment by taalexander** [comment]
Yes, this is a bug. 

I agree typing isn't properly supported, is there a negative to leaving it as is, or is there a different way to annotate? 

Do you think this is critical enough to require a backport, or is a normal bugfix fine?

**Comment by nonhermitian** [comment]
The negative is that it does not work, and you still need to do what would have always been done which is do `times = np.asarray(times)`.  I am personally against typing in Python, so others can decide.  I do not think it requires a backport.

**Comment by taalexander** [comment]
Typing is never strictly enforced, so there is no change there just an existing bug in the methods. It is only a `hint` and making it clear what is expected and can be algorithmically verified by offline tools.

**Comment by taalexander** [comment]
Closed via #2796.
